### PR TITLE
Added custom price to OrderLine.

### PR DESCRIPTION
**Pull request by KirillPlaksin**
I want to merge this change because it resolves https://github.com/saleor/saleor/issues/12777

The `OrderLineCreateInput` is extended with the optional price field `price`: PositiveDecimal, so the custom price could be set in the following mutation:
`DraftOrderCreate`
`orderLinesCreate`

# Impact

* [ ] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [x] Privileged queries and mutations are either absent or guarded by proper permission checks
* [x] Database queries are optimized and the number of queries is constant
* [x] Database migrations are either absent or optimized for zero downtime
* [x] The changes are covered by test cases

### CODE REVIEW

File: saleor/graphql/order/mutations/draft_order_create.py [hunk 1]
```diff
@@ -65,6 +65,14 @@ class OrderLineCreateInput(OrderLineInput):
             "by skipping the matching logic. " + ADDED_IN_36
         ),
     )
+    price = PositiveDecimal(
+        required=False,
+        description=(
+            "Custom price of the item."
+            "When the line with the same variant "
+            "will be provided multiple times, the last price will be used."
```
Comment (korycins) [inline saleor/graphql/order/mutations/draft_order_create.py:73]: You should add `ADDED_IN_314` + (probably) `PREVIEW_FEATURE` to all new fields that you're adding

File: saleor/graphql/order/types.py [hunk 2]
```diff
@@ -715,6 +715,11 @@ class OrderLine(ModelObjectType[models.OrderLine]):
             "permissions to access." + ADDED_IN_39
         ),
     )
+    price_override = graphene.Field(
```
Comment (korycins) [inline saleor/graphql/order/types.py:718]: Do you need this? I don't see that we're providing the `price_override` in `CheckoutLine` type

**Review by korycins — COMMENTED** [review]
[Question] Do we really need to have a `price_override` field on `OrderLine` DB model?
The flow is a little bit different than in Checkout. When order is created we're operating on the prices that are stored in `OrderLine` model.  So in case when we receive `price_override` from API, we just need to properly set  `undiscounted_base_unit_price` and `base_unit_price` with `price_override`. The all-price re-calculations logic uses this field (afair).

File: saleor/graphql/order/types.py (hunk 2 above)
Comment (IKarbowiak) [reply saleor/graphql/order/types.py:718]: Agree. It should be only on `CheckoutLine` model

File: CHANGELOG.md [hunk 3]
```diff
@@ -48,6 +48,7 @@ All notable, unreleased changes to this project will be documented in this file.
 - Add `ORDER_REFUNDED`, `ORDER_FULLY_REFUNDED`, `ORDER_PAID` webhooks - #12533 by @korycins
 - Handle error raised by 0Auth when fetching token - #12672 by @IKarbowiakg
 - Fix adding new lines to draft order when existing line has deleted product - #12711 by @SzymJ
+- Add `price_override` to `OrderLine` for set custom price in mutation `OrderLineCreateInput` - #12855 by @KirillPlaksin
```
Comment (IKarbowiak) [inline CHANGELOG.md:51]:
```suggestion
- Allow defining a custom price in draft orders - #12855 by @KirillPlaksin 
```

File: saleor/graphql/order/types.py (hunk 2 above)
Comment (KirillPlaksin) [reply saleor/graphql/order/types.py]: removed.

File: saleor/graphql/order/tests/mutations/test_draft_order_create.py [hunk 4]
```diff
@@ -1417,3 +1417,63 @@ def test_draft_order_create_with_non_unique_external_reference(
     assert error["field"] == "externalReference"
     assert error["code"] == OrderErrorCode.UNIQUE.name
     assert error["message"] == "Order with this External reference already exists."
+
+
+def test_draft_order_create_with_custom_price_in_order_line(
```
Comment (fowczarek) [inline saleor/graphql/order/tests/mutations/test_draft_order_create.py:1422]: We should add the similar test for unconfirmed orders. 
We should add tests with `force_new_line=True` for unconfirmed and draft orders.

Comment (KirillPlaksin) [reply saleor/graphql/order/tests/mutations/test_draft_order_create.py:1424]: https://github.com/saleor/saleor/pull/12855/files#diff-1de8a795a5a985ab55d370dbaa4f569ba1f15615bdd0195426de4d4d36c5290bR752

**Review by fowczarek — APPROVED** [review]

**Review by korycins — APPROVED** [review]

**Review by szczecha — APPROVED** [review]

**Pull request merged**